connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -80,7 +99,26 @@ class UCSingleCatalog extends TableCatalog with SupportsNamespaces with Logging
       // user-specified but system-generated, which is exactly the case here.
       newProps.put(TableCatalog.PROP_IS_MANAGED_LOCATION, "true")
       delegate.createTable(ident, columns, partitions, newProps)
+    } else if (hasLocationClause) {
+      val location = properties.get(TableCatalog.PROP_LOCATION)
+      assert(location != null)
+      val cred = temporaryCredentialsApi.generateTemporaryPathCredentials(
+        new GenerateTemporaryPathCredential().url(location).operation(PathOperation.PATH_READ))
```

Comment:
Should this be `PATH_CREATE_TABLE` here?
```suggestion
        new GenerateTemporaryPathCredential().url(location).operation(PathOperation.PATH_CREATE_TABLE))
```

Reply:
This is for CREATE EXTERNAL TABLE so the path already exists, Delta needs to read the delta log to do some validation, and I think this is `PATH_READ`?